man/systemd.exec.xml

```diff
@@ -1602,6 +1602,38 @@
         functions) if their standard output or standard error output is connected to the journal anyway, thus enabling
         delivery of structured metadata along with logged messages.</para></listitem>
       </varlistentry>
+
+      <varlistentry>
+        <term><varname>$SERVICE_RESULT</varname></term>
+
+        <listitem><para>Only defined for the service unit type, this environment variable is passed to all
+        <varname>ExecStop=</varname> and <varname>ExecStopPost=</varname> processes, and encodes the service
+        "result". Currently, the following values are defined: <literal>timeout</literal> (in case of an operation
+        timeout), <literal>exit-code</literal> (if a service process exited with a non-zero exit code; see
+        <varname>$EXIT_STATUS</varname> below for the actual exit status returned), <literal>signal</literal> (if a
+        service process was terminated abnormally by a signal; see <varname>$EXIT_STATUS</varname> below for the actual
+        signal used for the termination), <literal>core-dump</literal> (if a service process terminated abnormally and
+        dumped core), <literal>watchdog</literal> (if the watchdog keep-alive ping was enabled for the service but it
+        missed the deadline), <literal>start-limit-hit</literal> (the maximimum number of starts within the configured
+        time was reached), or <literal>resources</literal> (a catch-all condition in case a system operation
+        failed).</para></listitem>
+      </varlistentry>
+
+      <varlistentry>
+        <term><varname>$EXIT_CODE</varname></term>
+        <term><varname>$EXIT_STATUS</varname></term>
+
+        <listitem><para>Only defined for the service unit type, these environment variables are passed to all
+        <varname>ExecStop=</varname>, <varname>ExecStopPost=</varname> processes and contain exit status/code
+        information of the main process of the service. For the precise definition of the exit code and status, see
+        <citerefentry><refentrytitle>wait</refentrytitle><manvolnum>2</manvolnum></citerefentry>. <varname>$EXIT_CODE</varname>
+        is one of <literal>exited</literal>, <literal>killed</literal>,
+        <literal>dumped</literal>. <varname>$EXIT_STATUS</varname> contains the numeric exit code formatted as string
+        if <varname>$EXIT_CODE</varname> is <literal>exited</literal>, and the signal name in all other cases. Note
+        that these environment variables are only set if the service manager succeeded to start and identify the main
```

Comment:
> only set if the service manager succeeded to start

I think we should mention (explicitly) that this doesn't work for `Type=oneshot`

https://www.freedesktop.org/software/systemd/man/systemd.service.html#ExecStop=
> Note that the commands specified in ExecStop= are only executed when the service started successfully first.

```
-bash-4.3# systemctl cat self-abort --no-pager
# /etc/systemd/system/self-abort.service
[Service]
Type=oneshot
ExecStart=/bin/sh -x -c 'kill -ABRT $$$$'
ExecStop=/bin/sh -x -c 'set'

-bash-4.3# systemctl start self-abort
Job for self-abort.service failed because a fatal signal was delivered causing the control process to dump core.
See "systemctl status self-abort.service" and "journalctl -xe" for details.

-bash-4.3# journalctl -u self-abort --no-pager -o cat
self-abort.service: Collecting.
self-abort.service: Collecting.
self-abort.service: Collecting.
self-abort.service: Collecting.
self-abort.service: Trying to enqueue job self-abort.service/start/replace
self-abort.service: Installed new job self-abort.service/start as 380
self-abort.service: Enqueued job self-abort.service/start as 380
self-abort.service: About to execute: /bin/sh -x -c 'kill -ABRT $$$$'
self-abort.service: Forked /bin/sh as 298
self-abort.service: Changed dead -> start
Starting self-abort.service...
+ kill -ABRT 298
self-abort.service: Executing: /bin/sh -x -c 'kill -ABRT $$'
self-abort.service: Child 298 belongs to self-abort.service
self-abort.service: Main process exited, code=dumped, status=6/ABRT
self-abort.service: Changed start -> failed
self-abort.service: Job self-abort.service/start finished, result=failed
Failed to start self-abort.service.
self-abort.service: Unit entered failed state.
self-abort.service: Failed with result 'core-dump'.
self-abort.service: cgroup is empty
```